File: assignments/solutions/02-first-java/largestNumber.js
/**
 * Largest Number
 * 
 * @param {number} num1
 * @param {number} num2
 * @returns {number}
 */

const readline = require('readline');
const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

const largest = (num1, num2) => {
    return num1 > num2 ? num1 : num2;
}

console.log('Largest Number Program');

rl.question('Enter 1st number: ', num1 => {
    rl.question('Enter 2nd number: ', num2 => {
        if(num1 === num2)
            console.log('Equal Numbers');
        else
            console.log(`Largest Number: ${largest(num1, num2)}`);
        rl.close();
    });
});

/**
 * Output:
 *
 * Largest Number Program
 * Enter 1st number: 3
 * Enter 2nd number: 4
 * Largest Number: 4
 * 
 * Largest Number Program
 * Enter 1st number: 3
 * Enter 2nd number: 3
 * Equal Numbers
 */

File: assignments/solutions/02-first-java/largestNumber.java
// Largest Number program

import java.util.Scanner;

public class largestNumber {
    public static void main (String args[]) {
        Scanner scan = new Scanner(System.in);

        System.out.println("Largest Number");
        System.out.print("Enter 1st number: ");
        int num1 = scan.nextInt();
        System.out.print("Enter 2nd number: ");
        int num2 = scan.nextInt();

        if (num1 == num2)
            System.out.println("Equal Numbers");
        else {
            int result = num1 > num2 ? num1 : num2;
            System.out.println("Largest Number: " + result);
        }
    }
}

/*
Output:

Largest Number
Enter 1st number: 4
Enter 2nd number: 1
Largest Number: 4

Largest Number
Enter 1st number: 3
Enter 2nd number: 3
Equal Numbers
*/